frontend/alert/src/components/managealert/index.js

```diff
@@ -0,0 +1,189 @@
+                            ));
+                        const time = Intl.DateTimeFormat("en-US", {
+                            hour: "numeric",
+                            minute: "numeric",
+                            hour12: true,
+                        }).format(new Date(section.notification_sent_at));
+                        datetime = `${date} at ${time}`;
+                    }
+
+                    const status = section.section_status
+                        ? AlertStatus.Open : AlertStatus.Closed;
+
+                    let repeat;
+                    if (section.is_active) {
+                        if (section.auto_resubscribe) {
+                            repeat = AlertRepeat.EOS;
+                        } else {
+                            repeat = AlertRepeat.Once;
+                        }
+                    } else {
+                        repeat = AlertRepeat.Inactive;
+                    }
+
+                    return {
+                        id: section.id,
+                        original_created_at: section.original_created_at,
+                        section: section.section,
+                        date: datetime,
+                        status,
+                        repeat,
+                        actions: (repeat === AlertRepeat.Once || repeat === AlertRepeat.EOS)
+                            ? AlertAction.Cancel : AlertAction.Resubscribe,
+                    };
+                })
+            )));
+
+    AwesomeDebouncePromise(fetchPromise, 300)()
+        .then(alerts => setAlerts(alerts));
+};
+
+const filterAlerts = (alerts, filter) => {
+    const sortedAlerts = alerts.sort((a, b) => {
+        const d1 = new Date(a.original_created_at);
+        const d2 = new Date(b.original_created_at);
+        if (d1 > d2) {
+            // if d1 is later, a should come first
+            return -1;
+        }
+        return 1;
+    });
+
+    return sortedAlerts.filter((alert) => {
+        if (filter.search.length > 0) {
+            return alert.section.includes(filter.search.toUpperCase());
+        }
+        return true;
+    });
+};
+
+const getActionPromise = (id, actionenum) => {
+    let body;
+    switch (actionenum) {
+        case AlertAction.Resubscribe:
+            body = { resubscribe: true };
+            break;
+        case AlertAction.Cancel:
+            body = { cancelled: true };
+            break;
+        case AlertAction.Delete:
+            body = { deleted: true };
+            break;
+        default:
+    }
+
+    return fetch(`/api/alert/api/registrations/${id}/`, {
+        method: "PUT",
+        credentials: "include",
+        mode: "same-origin",
+        headers: {
+            Accept: "application/json",
+            "Content-Type": "application/json",
+            "X-CSRFToken": getCsrf(),
+        },
+        body: JSON.stringify(body),
+    });
+};
+
+
+const actionHandler = callback => (id, actionenum) => {
```

Comment:
This is basically a mini bit of redux here -- a descriptive comment would go a ways towards illuminating that for someone adding more functionality down the line.